R/03-sbc-centric-stats.R

```diff
@@ -0,0 +1,535 @@
+    dwellings,
+    households,
+    pct_of_csd
+  ) |>
+  left_join(
+    pop_projections |>
+      filter(year %in% c(CURRENT_YEAR, CURRENT_YEAR + 5, CURRENT_YEAR + 10)),
+    by = c("csd_clean" = "region"),
+    relationship = "many-to-many"
+  )
+
+# Preprocess db_projections data to transform age columns into rows
+# The columns x0, x1, etc. represent different age groups
+db_projections_transformed <- db_projections |>
+  # Get all column names that start with 'x' followed by digits (age columns)
+  pivot_longer(
+    cols = starts_with("x") & matches("^x\\d+$"),
+    names_to = "age_column",
+    values_to = "population_by_age"
+  ) |>
+  # Extract age values from column names (remove 'x' prefix)
+  mutate(
+    age = as.numeric(str_replace(age_column, "^x", "")),
+    # Create age groups (0-4, 5-9, etc.)
+    age_group = case_when(
+      age < 5 ~ "0-4",
+      age < 10 ~ "5-9",
+      age < 15 ~ "10-14",
+      age < 20 ~ "15-19",
+      age < 25 ~ "20-24",
+      age < 30 ~ "25-29",
+      age < 35 ~ "30-34",
+      age < 40 ~ "35-39",
+      age < 45 ~ "40-44",
+      age < 50 ~ "45-49",
+      age < 55 ~ "50-54",
+      age < 60 ~ "55-59",
+      age < 65 ~ "60-64",
+      age < 70 ~ "65-69",
+      age < 75 ~ "70-74",
+      age < 80 ~ "75-79",
+      age < 85 ~ "80-84",
+      age < 90 ~ "85-89",
+      TRUE ~ "90+"
+    ),
+    # Calculate population estimate for each DB
+    population = population_by_age * pct_of_csd,
+    total = total * pct_of_csd
+  )
+
+# this table now has a pct of csd column that we can use
+# to multiple by any other population projections to get estimates
+db_projections_transformed
+
+# =========================================================================== #
+# Metrics for SBC locations of interest ----
+# metrics of interest include:
+# - histograms or ridgelines of drive times/distances
+# - map that has all the 'closest' dbs
+# - demographic estimates at DB level
+# =========================================================================== #
+
+# first assign every dissemination block a 'nearest facility'
+assigned_facility <- drivetime_data |>
+  count(dbid, nearest_facility) |>
+  arrange(dbid, desc(n)) |>
+  group_by(dbid) |>
+  slice_head(n = 1) |>
+  rename(assigned = nearest_facility) |>
+  select(-n) |>
+  ungroup()
+
+# filter to only a couple of facilities
+drivetime_data <- drivetime_data |> left_join(assigned_facility, by = "dbid")
+
+drivetime_data_reduced <- drivetime_data |>
+  filter(
+    assigned %in%
+      (sbc_locs |>
+         filter(csd_name %in% CSD_NAMES) |>
+         pull(nearest_facility))
+  ) |>
+  filter(
+    nearest_facility %in%
+      (sbc_locs |>
+         filter(csd_name %in% CSD_NAMES) |>
+         pull(nearest_facility))
+  ) |>
+  left_join(crosswalk, by = c("dbid", "daid"))
+
+## SBC centric measures (table form) ----
+# drive measures
+drive_measures <- drivetime_data_reduced |>
+  group_by(assigned) |>
+  summarize(
+    n_addresses = n(),
```

Comment:
Consider adding a few na.rm = TRUE in these summary calculations, esp. given the previous left joins

Reply:
fixed in new file